Adafruit_CAP1188.cpp

```diff
@@ -63,24 +103,34 @@ Adafruit_CAP1188::Adafruit_CAP1188(int8_t clkpin, int8_t misopin,
   _i2c = false;
 }
 
-boolean Adafruit_CAP1188::begin(uint8_t i2caddr) {
+/*!
+ *    @brief  Setups the i2c depending on selected mode (I2C / SPI, Software / Hardware).
+ *            Displays useful debug info, as well as allow multiple touches (CAP1188_MTBLK), links leds to touches
+ *            (CAP1188_LEDLINK), and increase the cycle time value (CAP1188_STANDBYCFG) 
+ *    @param  i2caddr
+ *            optional i2caddres (default to 0x29)
+ *    @param  theWire
+ *            optional wire object
+ *    @return True if initialization was successful, otherwise false.
+ */
+boolean Adafruit_CAP1188::begin(uint8_t i2caddr, TwoWire *theWire) {
   if (_i2c) {
-    Wire.begin();
-    
+    _wire = theWire;
     _i2caddr = i2caddr;
+    _wire->begin();
   } else if (_clk == -1) {
     // Hardware SPI
     pinMode(_cs, OUTPUT);
     digitalWrite(_cs, HIGH);
 #ifdef SPI_HAS_TRANSACTION
-      SPI.begin();
+    SPI.begin();
 #else
```

Comment:
honestly we shouldn't support SPI with transaction anymore, please remove all sections that are not SPI_HAS_TRANSACTION